subsys/bluetooth/mesh/light_ctl_internal.h

```diff
@@ -17,19 +17,20 @@
 static inline uint16_t lvl_to_temp(struct bt_mesh_light_temp_srv *srv,
 				   int16_t lvl)
 {
-	return srv->range.min + (lvl + ((UINT16_MAX / 2) + 1)) *
-					(srv->range.max - srv->range.min) /
-					UINT16_MAX;
+	return srv->range.min +
+	       ROUNDED_DIV((int32_t)(lvl + ((UINT16_MAX / 2) + 1)) *
+				(srv->range.max - srv->range.min),
+				(int32_t)UINT16_MAX);
```

Comment:
This is looking very busy. Consider putting `((UINT16_MAX / 2) + 1)` in a define, such as `LIGHT_TEMP_LVL_OFFSET`. 

Also, its definition seems a bit weird. I think setting it to `0 - BT_MESH_LVL_MIN` makes more sense, it's accounting for the offset between LVL_MIN and LIGHT_TEMP_MIN, which is `BT_MESH_LVL_MIN` and `0`, respectively. Or flip it around, and make it `BT_MESH_LVL_MIN`.

Could also put `srv->range.max - srv->range.min` in a variable

Reply:
I tried to keep it as it was. Considering your comment, I think I will just use the formula as in specification with actual numbers, which keeps it simple and easy to follow.

Reply:
This seems like a degrade to me, tbh. `UINT16_MAX` communicated that the number was supposed to be the highest representable number, which is technically the intent in the spec as well